### PR TITLE
Update pip to generic http service

### PR DESCRIPTION
**Pull request by trescube**
Same deal as generic service in #850.

Configured as 

```json
"api": {
    "legacyUrl": "http://pelias.mapzen.com",
    "textAnalyzer": "libpostal",
    "services": {
      "pip": {
        "url": "http://localhost:3102",
        "timeout": 100,
        "retries": 1
      }
    }
  }
```

### CODE REVIEW

File: service/pointinpolygon.js [hunk 1]
```diff
@@ -1,86 +0,0 @@
-const logger = require( 'pelias-logger' ).get( 'pointinpolygon' );
```
Comment (dianashk) [inline service/pointinpolygon.js]: hooray for piles of deleted code!

**Comment by orangejulius** [comment]
I think the Joi schema config needs to be updated, and some error handling could be improved.

With a pelias config that looks like this, I get a schema error:
```json
 "api": {
    "services": {
      "placeholder": {
        "url": "http://internal-pelias-dev-placeholder-498151958.us-east-1.elb.amazonaws.com/parser"
      },
      "pip": {
        "url": "http://internal-pelias-dev-pip-1234964516.us-east-1.elb.amazonaws.com/"
      }
    },
    "textAnalyzer": "libpostal",
    "indexName": "pelias",
    "pipService": "http://internal-pelias-dev-pip-1234964516.us-east-1.elb.amazonaws.com/"
  },
```

The error is:
```
julian@julian-mapzen ~/repos/pelias/api $ npm start

> pelias-api@0.0.0-semantic-release start /home/julian/repos/pelias/api
> node index.js

/home/julian/repos/pelias/api/node_modules/pelias-config/index.js:29
      throw new Error(result.error.details[0].message);
      ^

Error: "pip" is not allowed
    at Object.generate (/home/julian/repos/pelias/api/node_modules/pelias-config/index.js:29:13)
    at Object.<anonymous> (/home/julian/repos/pelias/api/app.js:4:47)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/julian/repos/pelias/api/index.js:2:11)
```


If I _remove_ the PIP entry from services, I get the following error:
```
julian@julian-mapzen ~/repos/pelias/api $ npm start

> pelias-api@0.0.0-semantic-release start /home/julian/repos/pelias/api
> node index.js

loading libpostal data, this may take a few seconds...
Elasticsearch INFO: 2017-06-05T16:17:19Z
  Adding connection to http://internal-pelias-dev1-es2-838308460.us-east-1.elb.amazonaws.com:9200/

/home/julian/repos/pelias/api/node_modules/pelias-microservice-wrapper/ServiceConfiguration.js:12
    if (config.url && !_.endsWith(config.url, '/')) {
              ^

TypeError: Cannot read property 'url' of undefined
    at PointInPolygon.ServiceConfiguration (/home/julian/repos/pelias/api/node_modules/pelias-microservice-wrapper/ServiceConfiguration.js:12:15)
    at PointInPolygon (/home/julian/repos/pelias/api/service/configurations/PointInPolygon.js:11:5)
    at Object.addRoutes (/home/julian/repos/pelias/api/routes/v1.js:92:28)
    at Object.<anonymous> (/home/julian/repos/pelias/api/app.js:23:4)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
```

**Comment by trescube** [comment]
I'll update the schema.  Those commits should be just a weird artifact of a rebase but I'll look into it.

File: test/unit/schema.js [hunk 2]
```diff
@@ -547,14 +626,17 @@ module.exports.tests.api_services_validation = (test, common) => {
...
+module.exports.tests.pip_service_validation = (test, common) => {
+  test('when api.services.pip is defined, url is required', (t) => {
     var config = {
       api: {
         version: 'version value',
         indexName: 'index name value',
         host: 'host value',
         services: {
-          placeholder: {
+          pip: {
```
Comment (orangejulius) [inline test/unit/schema.js]: wait so are a bunch of placeholder tests actually changed to be pip tests or is the diff just interpreting the changes very weirdly?

Comment (trescube) [inline test/unit/schema.js]: I think it's interpreting the diff strangely.  Basically, there are placeholder and pip services schema tests that test the same functionality.

File: README.md [hunk 3]
```diff
@@ -45,7 +45,7 @@ The API recognizes the following properties under the top-level `api` key in you
 |`legacyUrl`|*no*||the url to redirect to in case the user does not specify a version such as `v1`
 |`relativeScores`|*no*|true|if set to true, confidence scores will be normalized, realistically at this point setting this to false is not tested or desirable
 |`accessLog`|*no*||name of the format to use for access logs; may be any one of the [predefined values](https://github.com/expressjs/morgan#predefined-formats) in the `morgan` package. Defaults to `"common"`; if set to `false`, or an otherwise falsy value, disables access-logging entirely.|
-|`pipService`|*yes*||full url to the pip service to be used for coarse reverse queries. if missing, which is not recommended, the service will default to using nearby lookups instead of point-in-polygon.|
+|`services`|*no*||service definitions for [point-in-polygon](https://github.com/pelias/pip-service) and [placholder](https://github.com/pelias/placeholder) services.  If missing (which is not recommended), the new point-in-polygon and placeholder services will not be called.|
```
Comment (orangejulius) [inline README.md]: i'd recommend just removing the word `new` now unless you want to determine a date after which they're no longer new and then promise to remember at that time :P

File: schema.js [hunk 4]
```diff
@@ -26,9 +26,14 @@ module.exports = Joi.object().keys({
     localization: Joi.object().keys({
       flipNumberAndStreetCountries: Joi.array().items(Joi.string().regex(/^[A-Z]{3}$/))
     }).unknown(false),
-    pipService: Joi.string().uri({ scheme: /https?/ }),
+    pipService: Joi.any().forbidden(), // got moved to services
```
Comment (orangejulius) [inline schema.js]: If you could, keep this change out for a little bit. That way we can leave the old `pipService` definition in the production config, but add the new one, and then transition over more smoothly. I think we could handle updating the API code and config at the same time, but its easier if we don't have to worry about it.

**Pull request merged**